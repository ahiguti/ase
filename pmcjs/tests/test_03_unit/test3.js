
Perl.Use("PObjTest");
var pobj = new Perl("PObjTest");

print("getprop: ", pobj.GetProperty('y'));
print("getprop: ", pobj['y']);
print("getprop_smart: ", pobj.y);
pobj.SetProperty('x', 99);
pobj.z = 123;
print("setprop: ", pobj.x);
print("setprop_smart: ", pobj.z);
print("invoke: ", pobj.Invoke('foo', 1, 2, 3));
print("invoke_smart: ", pobj.bar(194, 2, 3));
print("tostr: ", pobj.tostr());

print("enum1");
for (var i in pobj) {
  print("  k=" + i + " v=" + pobj[i]);
}
print("enum2");
pobj.baz = 100;
for (var i in pobj) {
  print("  k=" + i + " v=" + pobj[i]);
}
print("enum3");
pobj.SetProperty('fubar', 1200);
pobj['fubar'] = 1200;
for (var i in pobj) {
  print("  k=" + i + " v=" + pobj[i]);
}
