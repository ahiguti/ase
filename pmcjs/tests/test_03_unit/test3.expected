getprop: this is prop-y
getprop: this is prop-y
getprop_smart: this is prop-y
setprop: 99
setprop_smart: 123
invoke: 6
invoke_smart: 19423
tostr: (y val=this is prop-y) (x val=99) (z val=123) 
enum1
  k=y v=this is prop-y
  k=x v=99
  k=z v=123
enum2
  k=y v=this is prop-y
  k=baz v=100
  k=x v=99
  k=z v=123
enum3
  k=y v=this is prop-y
  k=fubar v=1200
  k=baz v=100
  k=x v=99
  k=z v=123
